src/pages/filter/common/dashboard/dashboard.store.ts

```diff
@@ -16,30 +18,92 @@ import {
   IFuncPropertyStatus,
   TPropertyStatus,
 } from '@service-providers/common'
-import { IExtendedTUnitGroups } from './dashboard.interfaces'
-import { getStartLayout } from './dashboard.utils'
+import { IExtendedTUnitGroup, ModifySet } from './dashboard.interfaces'
+import {
+  getLayoutOnSubTabHeightChange,
+  getLayoutOnTabHeightChange,
+  getNewTabLayout,
+  getSortedTabs,
+  getStartLayout,
+  getUpdatedLayout,
+} from './dashboard.utils'
 
 export class DashboardStore {
+  private static _TABS: string = 'dashboardMainTabs'
+  private static _LAYOUT: string = 'dashboardLayout'
+
+  private _inCharts: boolean = false
+  private _filterValue: string = ''
+
+  private _groups: IExtendedTUnitGroup[] = []
+
+  private _mainTabs: IExtendedTUnitGroup[] = []
+  private _spareTabs: IExtendedTUnitGroup[] = []
+  private _mainTabsLayout: Layout[] = []
+
   public viewType: ViewTypeDashboard = ViewTypeDashboard.List
   public page: GlbPagesNames | undefined
   public selectedEnumVariants: string[] = []
 
+  public get showInCharts(): boolean {
+    return this._inCharts
+  }
+
+  public get filterValue(): string {
+    return this._filterValue
+  }
+
+  public get groups(): IExtendedTUnitGroup[] {
+    return this._groups
+  }
+
+  public get mainTabs(): IExtendedTUnitGroup[] {
+    return this._mainTabs
+  }
+
+  public get spareTabs(): IExtendedTUnitGroup[] {
+    return this._spareTabs
+  }
+
+  public get mainTabsLayout(): Layout[] {
+    return this._mainTabsLayout
+  }
+
   constructor() {
     makeAutoObservable(this)
 
+    reaction(() => this.dashBoardQuery, this.reset)
+
     reaction(
-      () => this.dashBoardQuery,
+      () => this.mainTabs,
       () => {
-        this.toggleViewType(ViewTypeDashboard.List)
+        LocalStoreManager.write(DashboardStore._TABS, this.mainTabs)
+        LocalStoreManager.write(DashboardStore._LAYOUT, this.mainTabsLayout)
```

Comment:
I think it would be set individually for a certain dataset, so the third argument of dataset **name should** be passed here.

Reply:
Fixed